### PR TITLE
Update project dependencies and coding style fixes

### PR DESCRIPTION
**Pull request by ikas**
Some improvements:
- Updating Webshot dependencies to more recent versions
- Replaced npm with yarn
- Updated Node to version 11
- Added Travis integration
- Added coding style checks on pre-commit hooks (using Husky)
- Add readiness, liveness checks and resource allocation to k8s files

### CODE REVIEW

File: k8s/production/deployment.yaml [hunk 1]
```diff
@@ -68,6 +70,23 @@ spec:
 
         ports:
           - containerPort: 5000
+        readinessProbe:
+          httpGet:
+            scheme: HTTP
+            path: /healthcheck
+            port: 4100
```
Comment (tiagojsag) [inline k8s/production/deployment.yaml]: this will fail

File: k8s/production/deployment.yaml [hunk 2]
```diff
@@ -68,6 +70,23 @@ spec:
...
+          initialDelaySeconds: 30
+          timeoutSeconds: 5
+          periodSeconds: 15
+        livenessProbe:
+          httpGet:
+            scheme: HTTP
+            path: /healthcheck
+            port: 4100
```
Comment (tiagojsag) [inline k8s/production/deployment.yaml]: and so will this

File: k8s/staging/deployment.yaml [hunk 3]
```diff
@@ -67,6 +70,23 @@ spec:
 
         ports:
           - containerPort: 5000
+        readinessProbe:
+          httpGet:
+            scheme: HTTP
+            path: /healthcheck
+            port: 4100
```
Comment (tiagojsag) [inline k8s/staging/deployment.yaml]: fail

File: k8s/staging/deployment.yaml [hunk 4]
```diff
@@ -67,6 +70,23 @@ spec:
...
+          initialDelaySeconds: 30
+          timeoutSeconds: 5
+          periodSeconds: 15
+        livenessProbe:
+          httpGet:
+            scheme: HTTP
+            path: /healthcheck
+            port: 4100
```
Comment (tiagojsag) [inline k8s/staging/deployment.yaml]: and more fail

File: package.json [hunk 5]
```diff
@@ -49,12 +68,24 @@
     "koa-mount": "2.0.0",
     "koa-router": "^7.0.1",
     "koa-send": "^4.1.0",
+    "koa-simple-healthcheck": "^0.0.1",
+    "load-grunt-tasks": "^5.1.0",
```
Comment (tiagojsag) [inline package.json]: this is a dev dependency

Comment (ikas) [reply package.json]: Eslint complaied that this should be a project dependency, not a dev:
image

Anyway, I reset it as dev dependency!

Comment (tiagojsag) [reply package.json]: https://github.com/resource-watch/dataset/blob/08f08424362866dc8eb5c479b958ad93afc88c7c/app/Gruntfile.js#L4

**Pull request merged**